src/alpaca_parser.yrl

```diff
@@ -145,12 +150,36 @@ record_type_members -> record_type_member ',' record_type_members : ['$1' | '$3'
 record_type -> open_brace record_type_members close_brace : 
   #t_record{members='$2'}.
 
+module_qualified_type_name -> symbol '.' symbol:
+  {symbol, L, Mod} = '$1',
+  {symbol, _, Name} = '$3',
+  {module_qualified_type_name, L, Mod, Name}.
+
+module_qualified_type -> module_qualified_type_name :
+  {module_qualified_type_name, L, Mod, Name} = '$1',
+  #alpaca_type{
+     line = L,
+     module = list_to_atom(Mod),
+     name = {type_name, L, Name}}.
+
+module_qualified_type -> module_qualified_type_name type_expressions:
+  {module_qualified_type_name, L, Mod, Name} = '$1',
+  #alpaca_type{
+     line = L,
+     module = list_to_atom(Mod),
+     name = {type_name, L, Name},
+     vars = [V || {type_var, _, _}=V <- '$2']}.
```

Comment:
But don't we lose information if we don't register that "type int_option" is the type option.option instantiated with an int argument? Not just any value of any type would do, it should be specifically ints. Right now it looks like we drop that information on the floor. What am I missing?